### PR TITLE
Add IEPD and GE API clients

### PR DESCRIPTION
**Pull request by alexche-cchange**
This is only GEs, please use it, will add IEPDs tomorrow.

### CODE REVIEW

File: src/openepd/api/average_dataset/generic_estimate_sync_api.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    def get_by_open_xpd_uuid(self, uuid: str, with_response: Literal[True]) -> tuple[GenericEstimate, Response]: ...
+
+    @overload
+    def get_by_open_xpd_uuid(self, uuid: str, with_response: Literal[False] = False) -> GenericEstimate: ...
+
+    def get_by_open_xpd_uuid(
+        self, uuid: str, with_response: bool = False
+    ) -> GenericEstimate | tuple[GenericEstimate, Response]:
+        """
+        Get Generic Estimates by OpenEPD UUID.
```
Comment (alexlitvinenko) [inline src/openepd/api/average_dataset/generic_estimate_sync_api.py:46]:
```suggestion
        Get Generic Estimate by Open xPD UUID.
```

File: src/openepd/api/average_dataset/generic_estimate_sync_api.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+        :param uuid: OpenEPD UUID
```
Comment (alexlitvinenko) [inline src/openepd/api/average_dataset/generic_estimate_sync_api.py:48]: Open xPD UUID

File: src/openepd/api/average_dataset/generic_estimate_sync_api.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        :return: GE
+        :raise ObjectNotFound: if Generic Estimates is not found
+        """
+        response = self._client.do_request("get", f"/generic_estimates/{uuid}")
```
Comment (alexlitvinenko) [inline src/openepd/api/average_dataset/generic_estimate_sync_api.py:52]: You aren't converting `HTTPError` into `ObjectNotFound`.

File: src/openepd/api/average_dataset/generic_estimate_sync_api.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    def get_by_open_xpd_uuid(self, uuid: str, with_response: Literal[False] = False) -> GenericEstimate: ...
+
+    def get_by_open_xpd_uuid(
+        self, uuid: str, with_response: bool = False
+    ) -> GenericEstimate | tuple[GenericEstimate, Response]:
+        """
+        Get Generic Estimates by OpenEPD UUID.
+
+        :param uuid: OpenEPD UUID
+        :return: GE
```
Comment (alexlitvinenko) [inline src/openepd/api/average_dataset/generic_estimate_sync_api.py:49]: `with_response` is missing

File: src/openepd/api/average_dataset/generic_estimate_sync_api.py [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+        :raise ObjectNotFound: if Generic Estimates is not found
```
Comment (alexlitvinenko) [inline src/openepd/api/average_dataset/generic_estimate_sync_api.py:50]:
```suggestion
        :raise ObjectNotFound: if Generic Estimate is not found
```

File: src/openepd/api/average_dataset/generic_estimate_sync_api.py [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+    ) -> tuple[list[GenericEstimatePreview], Response]: ...
+
+    def list_raw(
+        self, page_num: int = 1, page_size: int = 10, with_response: bool = False
+    ) -> list[GenericEstimatePreview] | tuple[list[GenericEstimatePreview], Response]:
+        """
+        List generic estimates.
+
+        :param page_num: page number
+        :param page_size: page size
```
Comment (alexlitvinenko) [inline src/openepd/api/average_dataset/generic_estimate_sync_api.py:158]: with_response is missing

File: src/openepd/api/sync_client.py [hunk 7]
```diff
@@ -17,6 +17,7 @@
 
 from requests.auth import AuthBase
 
+from openepd.api.average_dataset.generic_estimate_sync_api import GenericEstimateApi
```
Comment (alexlitvinenko) [inline src/openepd/api/sync_client.py:20]:
```suggestion
from openepd.api.average_dataset.sync_api import GenericEstimateApi
```

**Review by alexlitvinenko — APPROVED** [review]

File: src/openepd/api/average_dataset/generic_estimate_sync_api.py (hunk 3 above)
Comment (alexche-cchange) [reply src/openepd/api/average_dataset/generic_estimate_sync_api.py:52]: This happens inside of client implementation: 

```
def _do_request...
        error_handler = self._http_error_handlers.get(response.status_code, None)
        if error_handler:
            response = error_handler(response, raise_for_status)

...

@staticmethod
def handle_not_found(response: Response, raise_for_status: bool) -> Response | None:

```

File: src/openepd/api/sync_client.py (hunk 7 above)
Comment (alexche-cchange) [reply src/openepd/api/sync_client.py:20]: I don't think this is necessary. 
The end-users would be using the Client as an interface to the library, and making a layer of re-imports here would add not much value. 
Or maybe I am missing a point?

**Review by corvis — APPROVED** [review]

File: src/openepd/api/sync_client.py (hunk 7 above)
Comment (alexlitvinenko) [reply src/openepd/api/sync_client.py:20]: Just for consistency with others.

**Pull request merged**